lib/info-extras.js

```diff
@@ -124,11 +124,37 @@ exports.getPublished = (body) => {
  * @return {Array.<Object>}
  */
 exports.getRelatedVideos = (body) => {
-  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': {"rvs":', '},');
+  let jsonStr = util.between(body, '\'RELATED_PLAYER_ARGS\': ', '\n');
+  let watchNextJson;
   try {
-    jsonStr = JSON.parse(jsonStr);
-  } catch (err) {
-    return [];
+    jsonStr = jsonStr.substring(0, jsonStr.length - 1)
+    jsonStr = JSON.parse(jsonStr)
+    watchNextJson = JSON.parse(jsonStr["watch_next_response"])
   }
-  return jsonStr.split(',').map((link) => qs.parse(link));
-};
+  catch (err) {
+    return []
+  }
+  let secondaryResults = watchNextJson.contents.twoColumnWatchNextResults.secondaryResults.secondaryResults.results
+  let recomendedVideos = [];
+  for (let result of secondaryResults) {
+    if (result.compactVideoRenderer !== undefined) {
```

Comment:
just `(result.compactVideoRenderer)` should be enough
```suggestion
    if (result.compactVideoRenderer) {
```